src/helper/service.cpp

```diff
@@ -65,6 +65,11 @@ void service::run(std::future<bool> &&exit_signal)
 {
     std::chrono::time_point<std::chrono::steady_clock> before{0s};
     std::future_status fs = exit_signal.wait_for(0s);
+    // TODO runtime_id will be send by the extension when it can get it from the
+    // profiler. When that happen, this wont be needed
+    if (rc_client_->get_service_identifier().runtime_id.empty()) {
```

Comment:
What prevented you from putting this inside the client itself? Perhaps I'm missing something.